apollo-federation/src/sources/connect/validation/mod.rs

```diff
@@ -147,23 +147,38 @@ pub fn validate(schema: Schema) -> Vec<Message> {
             });
         }
     }
-    let connect_errors = schema
+
+    let mut connect_errors: Vec<Message> = vec![];
+    schema
         .types
         .values()
-        .filter_map(|extended_type| match extended_type {
-            ExtendedType::Object(object) => Some(object),
-            _ => None,
-        })
-        .flat_map(|object| {
-            validate_object_fields(
+        .for_each(|extended_type| match extended_type {
+            ExtendedType::Object(object) => connect_errors.extend(validate_object_fields(
                 object,
                 &schema,
                 &connect_directive_name,
                 &source_directive_name,
                 &all_source_names,
-            )
+            )),
+            ExtendedType::Union(union_type) => connect_errors.push(Message {
+                code: Code::UnsupportedAbstractType,
+                message: "Abstract schema types, such as `union`, are not supported when using connectors. You can check out our documentation at https://go.apollo.dev/connectors/best-practices#abstract-schema-types-are-unsupported.".to_string(),
+                locations: Location::from_abstract_type_node(union_type, source_map, "union")
+                    .into_iter()
+                    .collect(),
+            }),
+            ExtendedType::Interface(interface) => connect_errors.push(Message {
+                code: Code::UnsupportedAbstractType,
+                message: "Abstract schema types, such as `interface`, are not supported when using connectors. You can check out our documentation at https://go.apollo.dev/connectors/best-practices#abstract-schema-types-are-unsupported.".to_string(),
+                locations: Location::from_abstract_type_node(interface, source_map, "interface")
+                    .into_iter()
+                    .collect(),
+            }),
```

Comment:
Maybe we should put all of this into a single function called `validate_type` or something? Rather than letting this function continue to grow.

Then we could go back to `flat_map` instead of `for_each`, which I like better 😁

Reply:
I made an update to pull this into a new function but def would like some input so we can get it back to `flat_map`